src/DataStructures/Tensor/Expressions/Evaluate.hpp

```diff
@@ -329,4 +368,58 @@ auto evaluate(const RhsTE& rhs_tensorexpression) {
                                 ~rhs_tensorexpression);
   return lhs_tensor;
 }
+
+/*!
+ * \ingroup TensorExpressionsGroup
+ * \brief Update a LHS tensor to the result of a RHS tensor expression that
+ * contains it
+ *
+ * \details See documentation for `evaluate` for basic functionality.
+ *
+ * `update` differs from `evaluate` in that `update` should be used when some
+ * LHS `Tensor` has been partially computed, and now we would like to update it
+ * with a RHS expression that contains it. In other words, this should be used
+ * when we would like to emulate assignment operations like `LHS +=`, `LHS -=`,
+ * `LHS *=`, etc.
+ *
+ * ### Example usage
+ * The equation \f$L_j = G R^{i}{}_{jl} S^{l}{}_{i} + H_j\f$ can be broken
+ * up into an initial call to `evaluate` to compute
+ * \f$R^{i}{}_{jl} S^{l}{}_{i}\f$, followed by a call to `update` to compute the
+ * rest of the expression:
+ * \code{.cpp}
+ * auto L = TensorExpressions::evaluate<ti_j>(
```

Comment:
could snippet from a test?

Reply:
Addressed in new fixup https://github.com/sxs-collaboration/spectre/pull/3842/commits/f4a8355967b0d42817858a11d8927ac71025017c